pycheribuild/config/compilation_targets.py

```diff
@@ -1025,8 +1033,12 @@ class BaremetalFreestandingTargetInfo(BaremetalClangTargetInfo):
     os_prefix: str = "baremetal-"
 
     @classmethod
-    def _get_compiler_project(cls) -> "type[BuildLLVMInterface]":
-        return typing.cast("type[BuildLLVMInterface]", SimpleProject.get_class_for_target_name("llvm", None))
+    def _get_compiler_project(cls, config: CheriConfig) -> "type[BuildLLVMInterface]":
+        if config.riscv_cheri_isa == RiscvCheriISA.STD:
+            llvm_target = SimpleProject.get_class_for_target_name("cheri-alliance-llvm", None)
```

Comment:
Won't this break things like building for arm since we only build the RISC-V and host backends in CHERI Alliance LLVM?

Reply:
Hmm I guess so, I was thinking about things like the picolibc target where aarch32 and aarch64 are supported and use CHERI LLVM.

Reply:
Maybe I should just change those to use upstream LLVM?